test/test_foreach.py

```diff
@@ -911,8 +911,58 @@ def test_outplace_with_invalid_grads(self, device, dtype, op):
         self.assertIsNotNone(sample.input[0].grad)
         self.assertIsNone(sample.input[1].grad)
 
+    @ops(
+        foreach_unary_op_db + foreach_binary_op_db + foreach_pointwise_op_db + foreach_lerp_op_db,
+        dtypes=OpDTypes.supported,
+        allowed_dtypes=(torch.float64, torch.complex128),
+    )
+    def test_inplace_forward_mode_AD(self, device, dtype, op):
+        if not op.supports_forward_ad:
+            self.skipTest("forward AD not supported")
+
+        # note(crcrpar): These patterns are to skip "sample"s that are incompatible with some
+        # in-place foreach functions' forward path, not forward mode AD.
+        # Raising / xfailing them would rather lose coverage since `sample_inputs` yields
+        # different Scalars and ScalarLists.
+        patterns_to_tolerate = "|".join([
+            _BOOL_SUB_ERR_MSG,
+            # Some non-unary in-place ops with scalar/scalarlist would error out if there's non float ones.
+            "can't be cast to the desired",
+            "clamp is not supported for complex types",
+            "Found dtype Double but expected ComplexDouble",
+        ])
```

Comment:
It still a little feels a little unsafe to me to just ignore all of these patterns. A more precise way of doing this with less risk of hiding any errors, is to be more specific about which ops fail and with what types of inputs, and just skip those inputs. 

I'd recommend creating a custom sample inputs that could be passed to the OpInfos of the ops concerned. This custom sample inputs function could just loop through the samples in the generic foreach samples inputs function (propagate the parameters) and then skip the ones that contain the inputs that are problematic for that op, e.g.:

```python
def sample_inputs_foreach_with_scalars_skipped(self, device, dtype, require_grad,...):
    # Note about what is problematic about scalars
    for sample in foreach_sample_inputs(
        device=device, dtype=dtype, requires_grad=requires, num_input_tensors=...
    ):
         if not problematic(sample):
             yield sample
```
If we wanted to be even more precise, we can create an error_inputs function that would assert that those ops indeed fail when those types of inputs are passed. You can see examples of these in `torch/testing/_internal/common_methods_invocations.py`.

Reply:
hmm, I think I was stupid when writing foreach sample inputs function but it's already complicated enough... https://github.com/pytorch/pytorch/blob/e274c2e4fd5f5039a4837ac9668cd564e2e50463/torch/testing/_internal/common_methods_invocations.py#L8086

For now, I'd like to handle things in this test method rather, as in c9f59b5702cb4c7f6a4c81f80bef57e455a3c8e4

Reply:
Sounds okay for now!